astroid/brain/brain_builtin_inference.py

```diff
@@ -911,6 +911,44 @@ def _infer_copy_method(
     raise UseInferenceDefault()
 
 
+def _is_str_format_call(node: nodes.Call) -> bool:
+    """Catch calls to str.format()."""
+    return (
+        isinstance(node.func, nodes.Attribute)
+        and node.func.attrname == "format"
+        and isinstance(node.func.expr, nodes.Const)
+        and isinstance(node.func.expr.value, str)
```

Comment:
This will only catch literals, not names, e.g.

```python
>>> x = 'yolo {}'
>>> x.format('12')
'yolo 12'
```

Is that a limitation with the way astroid normally infers calls?

Reply:
Perhaps better for a follow-up?